src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php

```diff
@@ -25,7 +25,7 @@ private function filterFilesByExtension(Traversable $files, string $fileExtensio
 
         /* @var $file \SplFileInfo */
         foreach ($files as $file) {
-            if (! preg_match($extensionMatcher, $file->getBasename())) {
+            if ($file->isDir() || ! preg_match($extensionMatcher, $file->getBasename())) {
```

Comment:
Should be checking for `! isFile` unless a particular decision was taken here to only exclude directories